flask_sqlalchemy/__init__.py

```diff
@@ -551,31 +555,99 @@ def get_engine(self):
 
 
 def _should_set_tablename(cls):
-    """Traverse the model's MRO. If a primary key column is found before a
-    table or tablename, then a new tablename should be generated.
+    """Check what values are set by a class' MRO to determine if a
+    tablename should be automatically generated.
+
+    Abstract classes do not generate a name, although they may have set
+    or inherited a name elsewhere.
+
+    If a class defines a tablename or table, a new one will not be generated.
+    Otherwise, if the class defines a primary key, a new name will be
+    generated.
+
+    This does not execute
+    :class:`sqlalchemy.ext.declarative.declared_attr` properties. If such an
+    attribute defines the primary key, decorate it with
```

Comment:
Is this also necessary when not using table name generation at all?

Reply:
No, nothing else would need to evaluate attrs prematurely except the tablename generation.